tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -361,17 +345,18 @@ ROOT::Experimental::Detail::RPage ROOT::Experimental::Detail::RPageSourceFile::P
 
    const auto element = columnHandle.fColumn->GetElement();
    const auto elementSize = element->GetSize();
-
    const auto bytesOnStorage = pageInfo.fLocator.fBytesOnStorage;
-   const auto bytesPacked = (element->GetBitsOnStorage() * pageInfo.fNElements + 7) / 8;
-   const auto pageSize = elementSize * pageInfo.fNElements;
 
-   auto pageBuffer = new unsigned char[bytesPacked];
+   const void *sealedPageBuffer = nullptr;
+   unsigned char *directReadBuffer = nullptr; // only used if cluster pool is turned off
    if (fOptions.GetClusterCache() == RNTupleReadOptions::EClusterCache::kOff) {
-      fReader.ReadBuffer(pageBuffer, bytesOnStorage, pageInfo.fLocator.fPosition);
+      directReadBuffer = new unsigned char[bytesOnStorage];
+      fReader.ReadBuffer(directReadBuffer, bytesOnStorage, pageInfo.fLocator.fPosition);
       fCounters->fNPageLoaded.Inc();
+      fCounters->fNRead.Inc();
+      fCounters->fSzReadPayload.Add(bytesOnStorage);
+      sealedPageBuffer = directReadBuffer;
```

Comment:
Those 2 counter updates seems new, if they are, should we mention them in the commit?

Reply:
Indeed, that was a bug fixed.  Added to the commit log.